packages/venia-ui/lib/util/apolloCache.js

```diff
@@ -63,3 +65,223 @@ export const cacheKeyFromType = object => {
             return defaultDataIdFromObject(object);
     }
 };
+
+const productKeyFieldFunction = object => {
```

Comment:
This is a cache key generating function. It does exactly what the old switch-case above did, but since I can't fall through object logic I defined it outside and used it below in the type policy. I'll try to make that clearer in name and by adding a comment.